src/index.ts

```diff
@@ -96,8 +100,12 @@ export class WaxJS {
     ) => void;
     metricURL?: string;
     returnTempAccounts?: boolean;
+    chainName?: string;
   }) {
-    this.rpc = new JsonRpc(rpcEndpoint);
+    this.registryRpc = new JsonRpc(rpcEndpoint);
+
+    this.setupChain(chainName);
```

Comment:
right you can't do await in a constructor. Plus it's a bad idea. Let's remove this setupChain call and replace it with
```
    this.chainName = chainName;
    this.chainId = chainId;
```

Reply:
updated 👍